CRM/Contribute/BAO/Contribution.php

```diff
@@ -3575,10 +3591,18 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
             $amount = $itemAmount;
             if (!$amount) {
               $amount = $params['total_amount'];
+              if ($context === NULL) {
+                $amount -= CRM_Utils_Array::value('tax_amount', $params, 0);
+              }
             }
           }
           else {
-            $amount = $diff * $fieldValues['line_total'];
+            if ($context == 'changedAmount') {
+              $itemParams['amount'] = $diff * ($fieldValues['line_total'] - $previousLineItem[$fieldValues['id']]['line_total']);
```

Comment:
so if this condition evaluated to true .. then the variable $amount will be undefined but it is still used when creating $itemParams array below .. I don't know what are you trying to do but this will cause issues so I think you need to rethink about what you are doing here